common/main/java/com/couchbase/lite/BasicAuthenticator.java

```diff
@@ -34,19 +36,37 @@
     // member variables
     //---------------------------------------------
 
+    @NonNull
     private final String username;
-    private final String password;
+    @NonNull
+    private final byte[] password;
 
     //---------------------------------------------
     // Constructor
     //---------------------------------------------
 
+    /**
+     * @param username
+     * @param password
+     * @deprecated Use BasicAuthenticator(String, byte[])
+     */
+    @Deprecated
     public BasicAuthenticator(@NonNull String username, @NonNull String password) {
-        Preconditions.assertNotNull(username, "username");
-        Preconditions.assertNotNull(password, "password");
+        this(username, Preconditions.assertNotNull(password, "password").getBytes(Charset.defaultCharset()));
+    }
 
-        this.username = username;
-        this.password = password;
+    /**
+     * Create a Basic Authenticator.
+     * The new instance contains a copy of the password byte array parameter:
+     * the owner of the original retains the responsiblity for zeroing it before releasing it.
+     *
+     * @return
+     */
+    public BasicAuthenticator(@NonNull String username, @NonNull byte[] password) {
```

Comment:
I noticed that Java's KeyStore use char[] as the parameter type for password. Do you think we should do the same?   The spec said byte[] but if char[] is preferred, we can do char[] as well.